packages/storage/amplify_storage_s3/example/integration_test/use_case_test.dart

```diff
@@ -726,6 +726,99 @@ void main() {
+              final fileNameTemp = 'userPrivate${uuid()}.txt';
+              await Amplify.Storage.uploadData(
+                data: S3DataPayload.bytes(
+                  testBytes,
+                  contentType: 'text/plain',
+                ),
+                key: fileKey,
+                options: StorageUploadDataOptions(
+                  accessLevel: accessLevel,
+                  metadata: {
+                    'filename': fileNameTemp,
+                  },
+                ),
+              ).result;
+            }
+
+            // Call list() and ensure length of result matches pageSize.
+            final listResult = await Amplify.Storage.list(
+              options: const StorageListOptions(
+                accessLevel: accessLevel,
+                pageSize: filesToList,
+              ),
+            ).result;
+            expect(listResult.nextToken?.isNotEmpty, isTrue);
+            expect(listResult.items.length, filesToList);
+            // Clean up files from this test.
+            await Amplify.Storage.removeMany(
+              keys: uploadedKeys,
+              options: const StorageRemoveManyOptions(
+                accessLevel: accessLevel,
+              ),
+            ).result;
+          });
+
+          testWidgets('list uses nextToken for pagination',
+              (WidgetTester tester) async {
+            const filesToUpload = 2;
+            const filesToList = 1;
+            const accessLevel = StorageAccessLevel.private;
+            final uploadedKeys = <String>[];
+            // Upload some files to test.
+            for (var i = filesToUpload; i > 0; i--) {
+              final fileKey = 'testObjectKey${uuid()}';
+              uploadedKeys.add(fileKey);
+              final fileNameTemp = 'userPrivate${uuid()}.txt';
+              await Amplify.Storage.uploadData(
+                data: S3DataPayload.bytes(
+                  testBytes,
+                  contentType: 'text/plain',
+                ),
+                key: fileKey,
+                options: StorageUploadDataOptions(
+                  accessLevel: accessLevel,
+                  metadata: {
+                    'filename': fileNameTemp,
+                  },
+                ),
+              ).result;
+            }
+
+            String? lastNextToken;
+            var timesCalled = 0;
+            while (lastNextToken != null || timesCalled == 0) {
+              // Call list() until nextToken is null and ensure we paginated expected times.
+              final listResult = await Amplify.Storage.list(
+                options: StorageListOptions(
+                  accessLevel: accessLevel,
+                  pageSize: filesToList,
+                  nextToken: lastNextToken,
+                ),
+              ).result;
+              lastNextToken = listResult.nextToken;
+              timesCalled++;
+            }
+            expect(timesCalled, greaterThanOrEqualTo(filesToUpload));
```

Comment:
Can we not be more precise?

Reply:
changed to put in specific dir for this test and can be exact